h/static/scripts/directives/via-dialog.coffee

```diff
@@ -0,0 +1,34 @@
+###*
+# @ngdoc directive
+# @name viaLink
+# @restrict A
+# @description The dialog that generates a via link to the page h is currently
+# loaded on.
+###
+
+viaLinkDialog = ['$timeout', ($timeout) ->
+    link: (scope, elem, attrs, ctrl) ->
+        ## Watch vialinkvisble: when it changes to true, focus input and selection.
+        scope.$watch (-> scope.viaLinkVisible), (visble) ->
+            if visble
+                $timeout (-> elem.find('#via').focus().select()), 0, false
+                
+    controller: ($scope, $element, $attrs) ->
+    	$scope.viaLinkVisible = false
+    	$scope.viaPageLink = 'https://via.hypothes.is/h/' + document.referrer
```

Comment:
A question here: Do we set the document.referrer when he inject the iframe? I'm just wondering if it'll always contain the uri. (I'd rather use the document plugin, but maybe it won't be needed)